pkg/reconciler/trigger/trigger.go

```diff
@@ -303,6 +303,27 @@ func (r *Reconciler) updateStatus(ctx context.Context, desired *v1alpha1.Trigger
 	return trig, err
 }
 
+// getBroker returns trigger's coupled broker. If the trigger is coupled with default broker, and default broker is not existed,
+// it will create the default broker first by labeling namespace with kantive-eventing-injection=enabled
+func (r *Reconciler) getBroker(ctx context.Context, t *v1alpha1.Trigger) (*v1alpha1.Broker, error) {
+	b, err := r.brokerLister.Brokers(t.Namespace).Get(t.Spec.Broker)
+	if t.Spec.Broker == "default" && apierrs.IsNotFound(err) {
+		current, _ := r.KubeClientSet.CoreV1().Namespaces().Get(t.Namespace, metav1.GetOptions{})
+		if current.Labels == nil {
+			current.Labels = map[string]string{}
+		}
+		current.Labels["knative-eventing-injection"] = "enabled"
+		_, e := r.KubeClientSet.CoreV1().Namespaces().Update(current)
+		if e != nil {
+			return b, err
+		}
+		// Wait for default broker to set up
+		time.Sleep(10 * time.Second)
```

Comment:
hmm this seems a bit random. What about checking the broker really exists, if not wait a bit (1 second) and retry, until it timeouts?

Reply:
I agree w/ @lionelvillard ... I was wondering too, about the 10 seconds guesstimate

Reply:
just synced offline with @grac3gao, she is working on the changes...
this https://github.com/knative/eventing/pull/2027 will probably have to land first in order to have this properly working... 

@grac3gao we can make an e2e integration test for this as well. But that can be done in a separate PR, and later


@grac3gao we can file a separate issue to include a e2e test for this as well. But that can be done later

Reply:
I'll remove the sleeping thing, and wait for #2027 landing. The purpose of this sleep is waiting for default broker to set up, otherwise, the trigger will still get the brokerNotExist error, even the broker is being created. Sync with @nachocano, it seems after the refactoring, the trigger will continue to reconcile after the broker is ready.